src/libostree/ostree-sign-ed25519.c

```diff
@@ -477,42 +489,46 @@ _ed25519_add_revoked (OstreeSign *self, GVariant *revoked_key, GError **error)
 }
 
 static gboolean
-_load_pk_from_stream (OstreeSign *self, GDataInputStream *key_data_in, gboolean trusted,
+_load_pk_from_stream (OstreeSign *self, GInputStream *key_stream_in, gboolean trusted,
                       GError **error)
 {
-  if (key_data_in == NULL)
+  if (key_stream_in == NULL)
     return glnx_throw (error, "ed25519: unable to read from NULL key-data input stream");
 
   gboolean ret = FALSE;
 
+  g_autoptr (OstreeBlobReader) blob_reader = ostree_sign_read_pk (self, key_stream_in);
+  g_assert (blob_reader);
+
   /* Use simple file format with just a list of base64 public keys per line */
   while (TRUE)
     {
-      gsize len = 0;
       g_autoptr (GVariant) pk = NULL;
       gboolean added = FALSE;
       g_autoptr (GError) local_error = NULL;
-      g_autofree char *line = g_data_input_stream_read_line (key_data_in, &len, NULL, &local_error);
+      g_autoptr (GBytes) blob = ostree_blob_reader_read_blob (blob_reader, NULL, &local_error);
 
       if (local_error != NULL)
         {
           g_propagate_error (error, g_steal_pointer (&local_error));
           return FALSE;
         }
 
-      if (line == NULL)
+      if (blob == NULL)
         return ret;
 
       /* Read the key itself */
-      /* base64 encoded key */
-      pk = g_variant_new_string (line);
+      pk = g_variant_new_fixed_array (G_VARIANT_TYPE ("y"), g_bytes_get_data (blob, NULL),
```

Comment:
Hmm isn't this...oh, I see `ostree_sign_ed25519_add_pk` already accepted both strings and byte arrays.